### PR TITLE
JBPM-5710 KIE Persistence Cleanup

### PR DESCRIPTION
**Pull request by marianbuenosayres**
This set of PRs is a set of modifications to allow multiple persistences without all of them depending on JPA:

- Creating drools-persistence-api and jbpm-persistence-api
- SingleSessionCommandService refactors
- PersistenceContext and ProcessPersistenceContext refactors:
- TaskPersistenceContext refactors
- Changes in the current persistence model interfaces
- TimerJobFactoryManager / TimerService refactors

### CODE REVIEW

File: kie-api/src/main/java/org/kie/api/persistence/ObjectStoringStrategy.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package org.kie.api.persistence;
```
Comment (mswiderski) [inline kie-api/src/main/java/org/kie/api/persistence/ObjectStoringStrategy.java]: missing license header

File: kie-api/src/build/revapi-config.json [hunk 2]
```diff
@@ -362,6 +362,32 @@
...
+        },
+        {
+          "code": "java.method.addedToInterface",
+          "new": "method java.lang.Boolean org.kie.api.task.model.Task::isArchived()",
+          "justification": "Moved from InternalTask to tidy up interfaces"
+        },
+        {
+          "code": "java.method.addedToInterface",
+          "new": "method <T> T org.kie.api.runtime.KieSession::getKieRuntime(java.lang.Class<T>)",
+          "justification": "Promoting internal method getKieRuntime() to the public API"
```
Comment (mswiderski) [inline kie-api/src/build/revapi-config.json]: is this really done here? can't find it in the commits...

Comment (marianbuenosayres) [inline kie-api/src/build/revapi-config.json]: @mswiderski that was actually from a manual conflict resolution I had to do on that file.

File: kie-api/src/main/java/org/kie/api/persistence/ObjectStoringStrategy.java (hunk 1 above)
Comment (marianbuenosayres) [inline kie-api/src/main/java/org/kie/api/persistence/ObjectStoringStrategy.java]: Added

**Comment by mswiderski** [comment]
@marianbuenosayres could you please squash all commits into one with jira commit message?

**Pull request merged**